main.go

```diff
@@ -1,7 +1,352 @@
 package main
 
-import "fmt"
+import (
+	"crypto/subtle"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"time"
 
+	"github.com/gorilla/mux"
+)
+
+// Certificate Struct
+type Certificate struct {
+	ID        string    `json:"id"`
+	Title     string    `json:"title"`
+	CreatedAt string    `json:"createdAt"` // Will be a type date
+	OwnerID   string    `json:"ownerId"`
+	Year      int       `json:"year"`
+	Note      string    `json:"note"`
+	Transfer  *Transfer `json:"transfer"`
+}
+
+// User Struct
+type User struct {
+	ID    string `json:"id"`
+	Email string `json:"email"`
+	Name  string `json:"name"`
+}
+
+// Transfer Struct
+type Transfer struct {
+	To     string `json:"to"`
+	Status string `json:"status"`
+}
+
+// Certificates
+var certificates []Certificate
+
+// Users
+var users []User
+
+// lookUpUserIDByEmail returns a the UsserID for a specified email address
+func lookUpUserIDByEmail(userEmail string) (string, error) {
+	for _, item := range users {
+		if item.Email == userEmail {
+			return item.ID, nil
+		}
+	}
+	return "", errors.New("Could not get user by email: " + userEmail)
+}
+
+// homehandler default route for testing
+func homeHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode("HOMEPAGE"); err != nil {
+		panic(err)
+	}
+}
+
+// initInMemoryData initiates some in memory data for testing
+func initInMemoryData() {
```

Comment:
We will want to delete this in memory init data and put it into unit tests. As a test we setup some fake data then call the handlers end point with that data and ensure we get the expected results back. Since the data in is always the same it allows us to have an automated way of ensuring our end points don't break. As our unit tests would fail and we could catch it before deploying.